src/Meditation.UI/ViewModels/ProcessListViewModel.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Immutable;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Meditation.Common.Models;
+using Meditation.Common.Services;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Meditation.UI.ViewModels
+{
+    public partial class ProcessListViewModel : ViewModelBase
+    {
+        private readonly IAttachableProcessListProvider processListProvider;
+
+        [ObservableProperty] private Task<ImmutableArray<ProcessInfo>> processList;
+        [ObservableProperty] private ProcessInfo? selectedProcess;
+        [ObservableProperty] private string? nameFilter;
+
+        public ProcessListViewModel(IAttachableProcessListProvider processListProvider)
+        {
+            this.processListProvider = processListProvider;
+            ProcessList = LoadAttachableProcessesAsync();
+        }
+
+        [RelayCommand]
+        public void FilterProcessList()
+        {
+            var filteredProcesses  = LoadAttachableProcessesAsync()
+                .ContinueWith(previous => previous.Result.Where(p => NameFilter == null || p.Name.Contains(NameFilter)).ToImmutableArray());
```

Comment:
[ToFix] Make a filtering type to do so. It'll help you alot when you'll need to implement such logic in other places like filter assemblies, types and methods.